### DISCUSSION TITLE
Allow providing excessive arguments when string formating

### DISCUSSION THREAD

**Discussion by timothyqiu** [godotengine/godot-proposals#5634] [Engine Core]
`"%d" % [1, 2, 3]` produces the string `"not all arguments converted during string formatting"`, I think it'll be more useful if it silently ignores the excessive arguments and produces `1`.

At least, it's better to return an empty string and print "not all arguments converted during string formatting" as error.

Allowing excessive arguments is useful, for example, when using `tr_n()`:

```gdscript
tr_n("%d day ago", "%d days ago", days) % days  # works

tr_n("one day ago", "%d days ago", days) % days # not allowed
```

**Comment by dalexeev** [comment]
The current workaround is to use the `String.format` method:

```gdscript
tr_n("{0} day ago", "{0} days ago", days).format([days])
tr_n("one day ago", "{0} days ago", days).format([days])

# or

tr_n("{days} day ago", "{days} days ago", days).format({days = days})
tr_n("one day ago", "{days} days ago", days).format({days = days})
```
